hyperspy/component.py

```diff
@@ -119,8 +120,50 @@ def __init__(self):
         self.component = None
         self.grad = None
         self.name = ''
+        self.units = ''
```

Comment:
Adding `units` to `Parameter` is probably also a good idea but one that requires its own PR...